crates/biome_js_analyze/src/analyzers/nursery/no_duplicate_else_if.rs

```diff
@@ -0,0 +1,215 @@
+    /// if-else-if chains are commonly used when there is a need to execute only one branch
+    /// (or at most one branch) out of several possible branches, based on certain conditions.
+    ///
+    /// Two identical test conditions in the same chain are almost always a mistake in the code.
+    /// Unless there are side effects in the expressions,
+    /// a duplicate will evaluate to the same true or false value as the identical expression earlier in the chain,
+    /// meaning that its branch can never execute.
+    ///
+    /// Please note that this rule does not compare conditions from the chain with conditions inside statements
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if (a) {
+    ///     foo();
+    /// } else if (b) {
+    ///     bar();
+    /// } else if (b) {
+    ///     baz();
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (a) {
+    ///     foo();
+    /// } else if (b) {
+    ///     bar();
+    /// } else if (c) {
+    ///     baz();
+    /// }
+    /// ```
+    ///
+    pub NoDuplicateElseIf {
+        version: "next",
+        name: "noDuplicateElseIf",
+        recommended: true,
+        source: RuleSource::Eslint("no-dupe-else-if"),
+    }
+}
+
+impl Rule for NoDuplicateElseIf {
+    type Query = Ast<JsIfStatement>;
+    type State = AnyJsExpression;
+
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let expr = node.test().ok()?;
+        let mut conditions_to_check: Vec<AnyJsExpression> = vec![];
+        conditions_to_check.push(expr.clone());
+
+        if let Some(logical_expression) = expr.as_js_logical_expression() {
+            if let Ok(operator_token) = logical_expression.operator() {
+                conditions_to_check.append(&mut split_by_logical_operator_wrapper(
+                    operator_token,
+                    &expr,
+                ));
+            }
+        }
+        let mut list_to_check: Vec<Vec<Vec<AnyJsExpression>>> = conditions_to_check
+            .iter()
+            .map(|c| {
+                split_by_logical_operator_wrapper(JsLogicalOperator::LogicalOr, c)
+                    .iter()
+                    .map(|f| split_by_logical_operator_wrapper(JsLogicalOperator::LogicalAnd, f))
+                    .collect()
+            })
+            .collect();
+        let mut current = node.syntax().clone();
+
+        while let Some(grand_parent_node) = current.grand_parent() {
+            current = grand_parent_node.clone();
+
+            let ifexpr = match grand_parent_node.kind() {
+                JsSyntaxKind::JS_IF_STATEMENT => {
+                    grand_parent_node.cast::<JsIfStatement>()?.test().ok()
+                }
+                _ => {
+                    break;
+                }
+            };
+            if let Some(expr) = ifexpr {
```

Comment:
This could be written more concisely:

```suggestion
            let Some(if_stmt) = grand_parent_node.cast::<JsIfStatement>() else {
                    break;
            };
            if let Ok(expr) = if_stmt.test() {
```